internal/runtime/client/client.go

```diff
@@ -549,3 +557,10 @@ func defaultDiscoveryResponse(discovery *runtimehooksv1.DiscoveryResponse) *runt
 	}
 	return discovery
 }
+
+// labelsForNamespace returns a labelSet containing the default LabelMetadataName for a namespace.
+func labelsForNamespace(namespace string) labels.Labels {
```

Comment:
This version works only with the namespace name. If we want it to also select base on labels applied to the namespace we'll have to retrieve the namespace object from the API server.

Reply:
I would say let's "Get" the Namespace to support the full featureset of the namespaceSelector. We can only retrieve partialobjectmeta which will make it more efficient (CR will also only cache metadata not the whole thing).

```go
	//   pod := &metav1.PartialObjectMetadata{}
	//   pod.SetGroupVersionKind(schema.GroupVersionKind{
	//       Group:   "",
	//       Version: "v1",
	//       Kind:    "Pod",
	//   })
	//   mgr.GetClient().Get(ctx, nsAndName, pod)
```

It's probably almost not worth the effort as a Namespace doesn't have a lot of data, but probably still a bit more efficient.


In any case I think we should support the full selector feature set